### PR TITLE
ISPN-5531 java.lang.UnsupportedOperationException during remove (using

### PR DESCRIPTION
**Pull request by wburns**
RemoteCacheManager)

* Batching event manager clears events on exception
* Distributed executor service now returns SuspectException

https://issues.jboss.org/browse/ISPN-5531

### CODE REVIEW

**Comment by wburns** [comment]
This should cherry-pick just fine into 7.2.x

File: core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java [hunk 1]
```diff
@@ -301,9 +301,18 @@ public void notifyCacheEntryCreated(K key, V value, boolean pre,
          EventImpl<K, V> e = EventImpl.createEvent(cache, CACHE_ENTRY_CREATED);
          configureEvent(e, key, value, pre, ctx, command, null, null);
          boolean isLocalNodePrimaryOwner = clusteringDependentLogic.localNodeIsPrimaryOwner(key);
-         for (CacheEntryListenerInvocation<K, V> listener : cacheEntryCreatedListeners) listener.invoke(e, isLocalNodePrimaryOwner);
-         if (!ctx.isInTxScope()) {
-            eventManager.sendEvents();
+         boolean sendEvents = !ctx.isInTxScope();
+         try {
+            for (CacheEntryListenerInvocation<K, V> listener : cacheEntryCreatedListeners)
+               listener.invoke(e, isLocalNodePrimaryOwner);
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java]: Missing braces

Comment (wburns) [inline core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java]: Ah yeah the original version didn't have them either :) I can add it in.

**Comment by wburns** [comment]
Updated.

**Comment by danberindei** [comment]
Integrated in master and 7.2.x, thanks Will!

**Pull request closed without merging**